mlx-mkbfb

```diff
@@ -594,6 +596,65 @@ def dump_stream(infn, do_dump, do_extract, prefix):
     inf.close()
 
 
+def strip_images(infn, outfn, version):
+    imgs = []
+
+    with open(infn, 'rb') as inf:
+        while True:
+            try:
+                inimg = Image(instream=inf)
+                imgs.append(inimg)
+            except NoHdr:
+                break
+
+    # Filter out version we don't want
+    imgs = [img for img in imgs if
+            img.get_image_ver() != version]
+
+    # Construct following images bitmap, and reduce it with each output.
+    fim_map = functools.reduce(
+        operator.or_,
+        (1 << img.get_image_id() for img in imgs)
+    )
+
+    # We won't enforce order here, but we need to know when to update
+    # fim map. Do this by grouping images by id, then taking the last one
+    # for each group. This is inefficient, but considering there's a max of 64
+    # image ids, shouldn't be too bad.
+    img_id_groups = [
+        [img for img in imgs if img.get_image_id() == id]
+        for _, _, id in image_list
+    ]
+    # Get last img for each group, filtering out empty groups.
+    last_imgs = [group[-1] for group in img_id_groups if group]
```

Comment:
Don't quite understand the logic here... why do we need the groups?
I mean, why not just get all the image IDs that needs to be removed, then go over the left images again to update the image map?

Reply:
Done, reworked this whole part to (hopefully) be a little clearer.